File: encoder8/Main.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Map<Character, Integer> vowelMapping = new HashMap<>();
        vowelMapping.put('a', 0);
        vowelMapping.put('e', 1);
        vowelMapping.put('i', 2);
        vowelMapping.put('o', 3);
        vowelMapping.put('u', 4);

        Scanner scanner = new Scanner(System.in);
        int number = scanner.nextInt();
        scanner.close();

        String decodedResult = decodeString(vowelMapping, number);
        System.out.println(decodedResult);
    }

    public static String decodeString(Map<Character, Integer> mapping, int encodedResult) {
        StringBuilder decodedStr = new StringBuilder();
        while (encodedResult > 0) {
            int index = encodedResult % 5;
            for (Map.Entry<Character, Integer> entry : mapping.entrySet()) {
                if (entry.getValue() == index) {
                    decodedStr.insert(0, entry.getKey());
                    break;
                }
            }
            encodedResult /= 5;
        }
        return decodedStr.toString();
    }
}

File: encoder8/main.py
def decode_string(mapping, encoded_result):
    decoded_str = ""
    while encoded_result > 0:
        index = encoded_result % 5
        for vowel, num in mapping.items():
            if num == index:
                decoded_str = vowel + decoded_str
                break
        encoded_result //= 5
    
    # Return the decoded string
    return decoded_str
    
vowel_mapping = {'a': 0, 'e': 1, 'i': 2, 'o': 3, 'u': 4}

# Example encoded result
number = int(input())

# Decode the encoded result using the provided mapping
decoded_result = decode_string(vowel_mapping, number)
print(decoded_result)